### DISCUSSION TITLE
Avoid extra branch creation

### DISCUSSION THREAD

**Discussion by violette** [renovatebot/renovate#9844] [Request Help] [answered]
Hello!

I need some help on my configuration for renovate... I'll try for a few days now and I'm am unbale to have the expected results.
First, it is unclear if the configuration must be in the renovate.json created by the first PR or not, so I created another one named renovate-config.json which I called when launching the docker image.

Here is my configuration:

```
{
  "endpoint": "<my_endpoint>",
  "repositories": [
    "~user/openig-renovate"
  ],
  "platform": "bitbucket-server",
  "onboardingConfig": {
    "extends": [
      "config:base"
    ],
    "prConcurrentLimit": 5
  },
  "packageRules": [{
    "excludePackagePatterns": [
      "^javax.servlet:javax.servlet-api"
    ]
  },
    {
      "managers": [
        "maven"
      ],
      "groupName": "Maven dependencies - debug label",
      "automerge": false
    },
    {
      "matchPaths": [
        "openig-ui/**"
      ],
      "groupName": "UI dependencies - debug label",
      "automerge": false
    }
  ],
  "separateMinorPatch": false,
  "separateMajorMinor": false,
  "prBodyColumns": [
    "Package",
    "Type",
    "Update",
    "New value",
    "References"
  ]
}

```

Renovate creates 5 PR and declines 3 named:
```
DEBUG: Branch lists (repository=~user/openig-renovate)
       "branchList": [
         "renovate/babel-monorepo",
         "renovate/major-maven-dependencies-debug-label",
         "renovate/major-react-dnd-monorepo",
         "renovate/major-react-monorepo",
         "renovate/major-ui-dependencies-debug-label",
         "renovate/maven-dependencies-debug-label",
         "renovate/react-monorepo",
         "renovate/ui-dependencies-debug-label"
       ],
       "renovateBranches": [
         "renovate/babel-monorepo", //declined just after creation - also contain a renovate.json file ?
         "renovate/major-maven-dependencies-debug-label", //declined just after creation - also contain a renovate.json file ?
         "renovate/maven-dependencies-debug-label", // open PR
         "renovate/react-monorepo", //declined just after creation - also contain a renovate.json file ?
         "renovate/ui-dependencies-debug-label" // open PR
       ]
```
It seems to works and I have only two open PR, see below, and it excludes the javax.servlet:javax.servlet-api.

If that possible, I'd like to achieve this:
- only 2 PR/branches created (one for the maven dependencies, one for the UI update : npm, eslint, etc, all located in the openig-ui folder), no extra branches creation
(Note that I delete all PR/branches between tests)

Any help appreciated,
Thanks!

**Comment by rarkins** [comment, marked as answer]
The `config:base` preset limits PR creation to 2 per hour

**Reply by violette** [reply]
Just tested it, even if I change the `renovate.json` and my config above with `"prConcurrentLimit": 2` I have the same result.
Branches are created then declined by renovate:
```
"renovateBranches": [
         "renovate/babel-monorepo", // modify files in openig-ui/npm-shrinkwrap.json and openig-ui/package.json (marked as declined just after created)
         "renovate/major-maven-dependencies-debug-label", // Update the javax.servlet-api.version even if marked excluded (marked as declined just after created)
         "renovate/maven-dependencies-debug-label", 
         "renovate/react-monorepo", // modify files in openig-ui/npm-shrinkwrap.json and package.json (marked as declined just after created)
         "renovate/ui-dependencies-debug-label"
       ]
```
I notice that the branch created and PR declined contain modifcation that are not present in open PRs.

I would like to have all the changes relatives to the UI in only one PR and one for the maven dependencies in another.

**Reply by rarkins** [reply]
Sorry but I don't fully understand what you're describing or what you're going to achieve. Maybe you just need package rules with grouping? If you are changing your config a lot then you could see some PRs get declined if they are no longer relevant. 

I suggest you start with a test repo with low number of dependencies until you have the grouping and rate limiting working as you'd like. It will make the logs more easy to understand too.

**Reply by violette** [reply]
Ok, I'm going to check that. Thanks
